plugins/kafka/kafka.cpp

```diff
@@ -1088,29 +1093,15 @@ ECL_KAFKA_API bool getECLPluginDefinition(ECLPluginDefinitionBlock* pb)
 
 MODULE_INIT(INIT_PRIORITY_STANDARD)
 {
-    KafkaPlugin::publisherCache = NULL;
-    KafkaPlugin::publisherCacheExpirer = NULL;
-
     return true;
 }
 
 MODULE_EXIT()
 {
     // Delete the background thread expiring items from the publisher cache
     // before deleting the publisher cache
-    if (KafkaPlugin::publisherCacheExpirer)
-    {
-        KafkaPlugin::publisherCacheExpirer->stop();
-        delete(KafkaPlugin::publisherCacheExpirer);
-        KafkaPlugin::publisherCacheExpirer = NULL;
-    }
-
-     if (KafkaPlugin::publisherCache)
-    {
-        KafkaPlugin::publisherCache->deleteAll();
-        delete(KafkaPlugin::publisherCache);
-        KafkaPlugin::publisherCache = NULL;
-    }
+    KafkaPlugin::publisherCacheExpirer.destroy();
```

Comment:
This still needs to call stop - simplest is in the destructor since nothing is derived from it.